src/smtsolvers/ScatterSplitter.cc

```diff
@@ -0,0 +1,593 @@
+        fprintf(stderr, "; |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
+        fprintf(stderr, "; ===============================================================================\n");
+    }
+    double next_printout = restart_first;
+
+    // Search:
+
+    if (config.dryrun())
+        stop = true;
+    while (status == l_Undef && !opensmt::stop && !this->stop)
+    {
+        // Print some information. At every restart for
+        // standard mode or any 2^n intervarls for luby
+        // restarts
+        if (conflicts == 0 || conflicts >= next_printout)
+        {
+            if ( config.verbosity() > 0 ) {
+                reportf("; %9d | %8d %8d | %8.3f s | %6.3f MB\n", (int) conflicts, (int) learnts.size(), nLearnts(),
+                        cpuTime(), memUsed() / 1048576.0);
+                fflush(stderr);
+            }
+        }
+
+        if (config.sat_use_luby_restart)
+            next_printout *= 2;
+        else
+            next_printout *= restart_inc;
+
+        // XXX
+        status = search((int)nof_conflicts, (int)nof_learnts);
+        nof_conflicts = restartNextLimit( nof_conflicts );
+        if (config.sat_use_luby_restart)
+        {
+            if (last_luby_k != luby_k)
+            {
+                nof_learnts *= 1.215;
+            }
+            last_luby_k = luby_k;
+        }
+        else
+        {
+            nof_learnts *= learntsize_inc;
+        }
+    }
+
+    if (status == l_True)
+    {
+        // Extend & copy model:
+        model.growTo(nVars());
+        for (int i = 0; i < nVars(); i++)
+            model[i] = value(i);
+    }
+    else
+    {
+        assert( opensmt::stop || status == l_False || this->stop);
+    }
+
+    // We terminate
+    return status;
+}
+
+lbool ScatterSplitter::zeroLevelConflictHandler() {
+    if (splits.size() > 0) {
+        opensmt::stop = true;
+        return l_Undef;
+    } else {
+        return CoreSMTSolver::zeroLevelConflictHandler();
+    }
+}
+
+bool ScatterSplitter::scatterLevel()
+{
+    int d;
+    if (!splitConfig.split_on) return false;
+    // Current scattered instance number i = splits.size() + 1
+    float r = 1/(float)(splitConfig.split_num-splits.size());
+    for (int i = 0; ; i++)
+    {
+        // 2 << i == 2^(i+1)
+        if ((2 << (i-1) <= splitConfig.split_num - static_cast<int>(splits.size())) &&
+            (2 << i >= splitConfig.split_num - static_cast<int>(splits.size())))
+        {
+            // r-1(2^i) < 0 and we want absolute
+            d = -(r-1/(float)(2<<(i-1))) > r-1/(float)(2<<i) ? i+1 : i;
+            break;
+        }
+    }
+    return d == decisionLevel()+assumptions.size();
+}
+
+
+bool ScatterSplitter::createSplit_scatter()
+{
+    assert(splits.size() == split_assumptions.size());
+    splits.emplace_back(SplitData(config.smt_split_format_length() == spformat_brief));
```

Comment:
It looks a little convoluted that we emplace first at the end the `SplitData`, and then filled through a reference.  I wonder if we could change it so that we `emplace_back` only at the end of the function.